docs/explanation/core/proposer-based-timestamps.md

```diff
@@ -55,44 +33,176 @@ The following protocols and application features require a reliable source of ti
+worst-case for the clock drift among all participants.
+Due to the [leap second events](https://github.com/tendermint/tendermint/issues/7724),
+it is recommended to use set `Precision` to at least `500ms`.
+
+The `Precision` parameter is given in milliseconds.
+
+### `SynchronyParams.MessageDelay`
+
+The `MessageDelay` parameter configures the acceptable upper-bound for the
+end-to-end delay for transmitting a `Proposal` message from the proposer to
+_all_ validators in the network.
+
+Networks should choose a `MessageDelay` that is large enough to represent the
+worst-case delay for a message to reach all participants.
+This delay does not depend, a priori, on the size of proposed blocks, since it
+applies to the `Proposal` messages.
+It does depend on the number of participants and latency of their connections.
+
+The `MessageDelay` parameter is given in milliseconds.
+
+### `FeatureParams.PbtsEnableHeight`
+
+The `PbtsEnableHeight` parameter configures the first height at which the PBTS
+algorithm should be adopted for generating and validating block timestamps in a network.
+In heights up to `PbtsEnableHeight - 1`, the generation and validation of block
+timestamps in the network will be carried out using the legacy BFT Time algorithm.
+
+While `PbtsEnableHeight` is set to `0`, the network will adopt the legacy BFT
+Time algorithm.
+
+When `PbtsEnableHeight` is set to a height `H > 0`, the network will switch to
+the PBTS algorithm from height `H`.
+The network will still adopt the legacy BFT Time algorithm to produce and
+validate block timestamps until height `H - 1`.
+The enable height `H` must be a future height, i.e., larger than the current
+blockchain height.
+
+Once `PbtsEnableHeight` is set and the PBTS algorithm is enabled (i.e., from height
+`PbtsEnableHeight`), it is not possible to return to the legacy BFT Time algorithm.
+The switch to PBTS is therefore irreversible.
+
+Finally, if `PbtsEnableHeight` is set `1` in the genesis file or by the application
+upon the ABCI `InitChain` method, the network will adopt PBTS from the first
+height. This is the recommended setup for new chains.
+
+The `PbtsEnableHeight` parameter is an integer.
+
+## Important Notes
+
+When configuring a network to adopt the PBTS algorithm, the follow steps must be considered:
+
+1. Make sure that the clocks of validators are [synchronized](#clock-synchronization) **before** enabling PBTS.
+1. Make sure that the configured value for [`SynchronyParams.Precision`](#synchronyparamsprecision) is
+   reasonable.
+1. Make sure that the configured value for [`SynchronyParams.MessageDelay`](#synchronyparamsmessagedelay) is
+   reasonable and large enough to reflect the worst-case delay for messages in the network.
+   Setting this parameter to a small value may impact the progress of the
+   network, namely blocks may take very long to be committed.
+1. Make sure that the block times **currently** produced by the network do not
+   differ too much from real time.
```

Comment:
Could rephrase to mention that this is only relevant if the BFT time is in the future, wrt to real time.

Reply:
Added.